Assets/Scripts/FlashLight.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FlashLight : MonoBehaviour
+{
+    private InputHandler input;
+    [SerializeField] private GameObject FLObject, FLModel;
+    [SerializeField] private float autoAimMinDistance = 1f;
+    [SerializeField] private float followDelay = 0.2f;
+    [SerializeField] private Camera MCamera;
+    [SerializeField] private LayerMask aimMask;
+
+    [SerializeField] private Light lightSource;
+    [SerializeField] private float MinAngle, MaxAngle;
+    [SerializeField] private float maxIntensity = 6f;
+    [SerializeField] private float minIntensity = 2f;
+    private float Charge = 120f;
+    private Animator anim;
+    private bool isOn = false;
+    
+    void Start()
+    {
+        input = InputHandler.Instance;
+        float t = (lightSource.innerSpotAngle - MinAngle) / (MaxAngle - MinAngle);
+        lightSource.intensity = Mathf.Lerp(maxIntensity, minIntensity, t);
+        anim = FLObject.GetComponent<Animator>();
+    }
+
+    void Update()
+    {
+        TurnOnOffLight();
+        ConfigureRadius();
+    }
+
+    private void FixedUpdate()
+    {
+        DecreaseCharge();
+    }
+
+    private void LateUpdate()
+    {
+        if (isOn)
+        {
+            AutoAim();
+        }
+
+    }
+    private void TurnOnOffLight()
+    {
+        if (input.FlashlightInput)
+        {
+            if (!isOn && Charge > 0f)
+            {
+                lightSource.enabled = true;
+                anim.SetBool("IsOn", true);
+                isOn = true;
+            }
+            else
+            {             
+                lightSource.enabled = false;
+                anim.SetBool("IsOn", false);
+                isOn = false;
+            }
+        }
+    }
+
+    private void AutoAim()
+    {
+        Quaternion targetRotation;
+        RaycastHit hit;
+        if (Physics.Raycast(MCamera.transform.position, MCamera.transform.forward, out hit, 20f, aimMask))
```

Comment:
the 20f is called a "magic number", ideally have a const at the top of the file for any hard coded values like this that could easily be tweaked in the future.